tools/src/test/java/org/apache/kafka/tools/AbstractResetIntegrationTest.java

```diff
@@ -406,6 +408,7 @@ protected boolean tryCleanGlobal(final boolean withIntermediateTopics,
         final Properties cleanUpConfig = new Properties();
         cleanUpConfig.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 100);
         cleanUpConfig.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, Integer.toString(CLEANUP_CONSUMER_TIMEOUT));
+        cleanUpConfig.put(ConsumerConfig.GROUP_PROTOCOL_CONFIG, GroupProtocol.CLASSIC.name);
```

Comment:
I am actually wondering if this is the right fix? It seem, we should change `StreamsResetter` and hard-code the "classic" protocol there (and guard it from being overwritten by the user).

Reply:
I see. Definitely sounds sensible to add a validation for the protocol in the `StreamsResetter` tool, and set it to classic there too (I can do it in a follow-up PR, let's just agree on what exactly you guys think it's best for the resetter tool experience)

I agree with your suggestion, only wondering if we should "guard it from being overwritten"? In cases like this test example, that sets properties that are strictly for the classic-protocol, it seems expected that one would naturally set the protocol to classic, along with the HB and session. If we add a guard, the user would get an error, which will seem noisy right? It's definitely a config not needed but it's not wrong. Thoughts?

Reply:
> I see. Definitely sounds sensible to add a validation for the protocol in the StreamsResetter tool, and set it to classic there too (I can do it in a follow-up PR, let's just agree on what exactly you guys think it's best for the resetter tool experience)

SG. I actually believe/hope, if we fix this inside `StreamsResetter` we might not need to set the protocol here and can revert these changes in `AbstractResetIntegrationTest` -- the fact that we need this changes here, seems to surface the actual bug inside `StreamsResettter` and if we fix this bug, we won't need it in the test setup any longer.

> wondering if we should "guard it from being overwritten"

The guard should be to disallow setting it to "consumer"... If one set's "classic" expliclity, we don't need to do anything.

> In cases like this test example, that sets properties that are strictly for the classic-protocol, it seems expected that one would naturally set the protocol to classic, along with the HB and session.

In the end, this should not be necessary to begin with.

> If we add a guard, the user would get an error, which will seem noisy right? It's definitely a config not needed but it's not wrong. Thoughts?

I don't think we need to raise an error. If one sets it to "classic" we can just accept it (ie, ignore it). And if one set's it to "consumer" we should overwrite to "classic" and log a warning.

Reply:
perfect, we're on the same page then, guarding against CONSUMER (not against setting CLASSIC explicitly). And agree on removing the unneeded line in the test. Will open a follow-up PR.

Reply:
follow-up for StreamsResetter https://issues.apache.org/jira/browse/KAFKA-18224